lib/service/service.go

```diff
@@ -1252,31 +1256,37 @@ func (process *TeleportProcess) getLocalAuth() *auth.Server {
 	return process.localAuth
 }
 
-func (process *TeleportProcess) setInstanceClient(clt *auth.Client) {
+func (process *TeleportProcess) setInstanceConnector(conn *Connector) {
 	process.Lock()
-	process.instanceClient = clt
+	process.instanceConnector = conn
 	process.Unlock()
-	process.instanceClientReadyOnce.Do(func() {
-		close(process.instanceClientReady)
+	process.instanceConnectorReadyOnce.Do(func() {
+		close(process.instanceConnectorReady)
 	})
 }
 
-func (process *TeleportProcess) getInstanceClient() *auth.Client {
+func (process *TeleportProcess) getInstanceConnector() *Connector {
 	process.Lock()
 	defer process.Unlock()
-	return process.instanceClient
+	return process.instanceConnector
 }
 
-// waitForInstanceClient waits for the instance client to become available. Instance client is only available if at least
-// one non-auth service is registered.  Auth-only instances cannot use the instance client because auth servers need to
-// be able to fully initialize without a valid CA in order to support HSMs.
-func (process *TeleportProcess) waitForInstanceClient() (clt *auth.Client, ok bool) {
+func (process *TeleportProcess) getInstanceClient() *auth.Client {
```

Comment:
Maybe add a godoc here indicating that this can return nil (and under what circumstances)